src/index.c

```diff
@@ -734,8 +734,11 @@ void git_index_entry__init_from_stat(
 {
 	entry->ctime.seconds = (git_time_t)st->st_ctime;
 	entry->mtime.seconds = (git_time_t)st->st_mtime;
-	/* entry->mtime.nanoseconds = st->st_mtimensec; */
-	/* entry->ctime.nanoseconds = st->st_ctimensec; */
+#if !defined(GIT_WIN32) && !defined(__APPLE__)
+	/* Apple and Windows doesn't provide these struct stat fields. */
```

Comment:
Ah, yep, you're right. The BSD's seem to support these fields (at least, [FreeBSD](https://www.freebsd.org/cgi/man.cgi?query=stat&sektion=2) and [OpenBSD](http://www.openssh.com/cgi-bin/man.cgi/OpenBSD-current/man2/stat.2?query=stat&sec=2)), however [Solaris doesn't seem to](http://docs.oracle.com/cd/E19109-01/tsolaris8/816-1056/6m7gi1i51/index.html). Perhaps we should enable the USE_NSEC option based upon some CMake feature check determining that these `struct stat` members are present?